apis/openstack-nova/src/main/java/org/jclouds/openstack/nova/v2_0/compute/extensions/NovaSecurityGroupExtension.java

```diff
@@ -191,14 +191,20 @@ public boolean removeSecurityGroup(String id) {
          return false;
       }
 
-      if (sgApi.get().get(groupId) == null) {
-         return false;
+      // Would be nice to delete the group and invalidate the cache atomically - i.e. use a mutex.
+      // Will make sure that a create operation in parallel won't see inconsistent state.
+
+      boolean deleted = sgApi.get().delete(groupId);
```

Comment:
The return value is false. This code path is covered in the test `testSecurityGroupCacheInvalidatedWhenDeletedExternally`